pkg/scale/varying_data_type_nested_example_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package scale_test
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// ParentVDT is a VaryingDataType that consists of multiple nested VaryingDataType
+// instances (aka. a rust enum containing multiple enum options)
+type ParentVDT scale.VaryingDataType
```

Comment:
Is index receiver not needed for the ParentVDT because you attempt to case to a VDT while decoding, and if you can then call custom method?

Reply:
index receiver isn't required for `ParentVDT` since this type isn't used as a `VaryingDataTypeValue` anywhere.